internal/idp/providers/saml/session.go

```diff
@@ -67,6 +68,9 @@ func (s *Session) FetchUser(ctx context.Context) (user idp.User, err error) {
 
 	s.Assertion, err = s.ServiceProvider.ServiceProvider.ParseResponse(s.Request, []string{s.RequestID})
 	if err != nil {
+		if invalidRespErr, ok := err.(*saml.InvalidResponseError); ok {
+			logging.OnError(invalidRespErr.PrivateErr).Info("invalid SAML response details")
+		}
```

Comment:
```suggestion
		invalidRespErr := new(saml.InvalidResponseError)
		if errors.As(err, &invalidRespErr) {
			logging.WithError(invalidRespErr.PrivateErr).Info("invalid SAML response details")
		}
```

Reply:
while your implementation would work, I suggest using the error.As function to go with best practice and to handle the case where the library would wrap the error

Reply:
Thanks for the suggestion, updated 👍